linsolve/linsolve.py

```diff
@@ -246,8 +263,8 @@ def __init__(self, data, wgts={}, sparse=False, **kwargs):
         #go through and figure out if any variables are conjugated
         for eq in self.eqs: 
             self.re_im_split |= eq.has_conj
-        numerical_input = list(self.data.values()) + list(self.consts.values()) + list(self.wgts.values())
-        self.dtype = reduce(np.promote_types, [d.dtype if hasattr(d,'dtype') else type(d) for d in numerical_input])
+        self.dtype = infer_dtype(list(self.data.values()) + list(self.consts.values()) + list(self.wgts.values()))
+        self.is_single = self.dtype.name in ['float32', 'complex64']
```

Comment:
Do we need these? `is_single` only seems to get assigned in the code but not used. I'd consider getting rid of these lines or making them `self._is_single` if it's there purely for diagnostic purposes for the advanced user.

Reply:
Removed in lieu of LinearSolver.dtype